go/json2/unmarshal.go

```diff
@@ -53,3 +53,9 @@ func annotate(data []byte, err error) error {
 
 	return fmt.Errorf("line: %d, position %d: %v", line, pos, err)
 }
+
+// UnmarshalPB is similar to Unmarshal but specifically for proto.Message to add type safety.
```

Comment:
Let's call `UnmarshalPB` from `Unmarshal` after the type check to remove the duplicate options we have now?

Reply:
done